scripts/shellcheck.sh

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env bash
+
+CYAN="\e[36m"
+YELLOW="\e[33m"
+BLUE="\e[34m"
+RED="\e[31m"
+GREEN="\e[32m"
+RST_COLOR="\e[0m"
```

Comment:
I'm always a bit hesitant to use color in shell output. If nothing else, I think it has to be used sparingly, and this strikes me as a bit excessive. Is there a specific motivation for having such colorized output?

Reply:
I find its much easier to understand the key insights quickly with the color. A friend who works in UI actually recommended I add it. We do have color output in some of the other scripts such as `./scripts/install-build-tools.sh` and `./scripts/setup-dependencies.sh`
failing:
![shellcheck-output](https://github.com/user-attachments/assets/94d1fa91-672c-4efc-97b4-640885eb47aa)
passing:
![shellcheck-pass](https://github.com/user-attachments/assets/215c9ab5-5cac-4cd0-a099-7caadf50d8f5)
The shellcheck report itself `./shellcheck-report.txt` does not have color.

Reply:
The report can have color, because `shellcheck` supports configurable colorization.

Other scripts do have color, and we should unify them at some point to behave well for accessibility reasons (color can be helpful, but it can also be terrible; the most accessible option is to support configuration). However, most of those are either run-once, run-super-rarely, or are intended for pipelines and not really for humans (like the end-to-end kubernetes test). If we want to support colorization in this tool, we should at least support a basic way of disabling/enabling it (as `shellcheck` does).

The gold standard is to offer `-c|--color[=WHEN]` where `WHEN` can be one of `auto` (which disables color when the output is going to a pipe or file, and is a bit tricky to really get right), `always`, and `never` (defaulting to `auto`); and, to support the [`NO_COLOR`](https://no-color.org/) environment variable. We should, at least, support the env-var if not something more flexible.

Reply:
ACK - gave this a shot